packages/fx-core/src/core/middleware/contextLoader.ts

```diff
@@ -51,6 +52,8 @@ export async function loadSolutionContext(tools: Tools, inputs: Inputs):Promise<
     const projectSettings: ProjectSettings = await fs.readJson(settingsFile);
     if(!projectSettings.currentEnv)
       projectSettings.currentEnv = "default";
+    if(!projectSettings.projectId)
+      projectSettings.projectId = uuid.v4();
```

Comment:
when will it be written to env json file?

Reply:
in ConfigWriterMW